src/spec.md

```diff
@@ -0,0 +1,335 @@
+| upload       | Asynchronously upload data of arbitrary size                   |
+| downloadSync | Synchronously download data                                    |
+| download     | Asynchronously download data                                   |
+| setDefault   | Set the default file for FlatDirectory                         |
+
+
+
+
+<p id="EthStorage"></p>
+
+## 3. EthStorage Class
+
+### Constructor
+
+**Description**: Create an instance of `EthStorage`.
+
+**Parameters**
+- `config` (object): Configuration object containing necessary settings.
+   - `rpc` (string): RPC for any evm network.
+   - `ethStorageRpc` (string): The EthStorage network rpc corresponding to this evm network, the data is obtained from the EthStorage network.
+   - `privateKey` (string): Wallet private key.
+
+**Example**
+```javascript
+const config = {
+   rpc: "your_rpc",
+   ethStorageRpc: "ethstorage_rpc",
+   privateKey: "your_private_key"
+};
+
+const ethStorage = new EthStorage(config);
+```
+
+
+
+### Methods
+
+#### estimateCost
+
+**Description**: Estimate the cost of uploading data.
+
+**Parameters**
+- `key` (string): The key for the data to be written.
+- `data` (Buffer): The data to be written, its size cannot exceed the maximum value of the content that can be transferred by a blob.
+
+**Returns**
+- `cost` (Promise<object>): A Promise that resolves to an object containing:
+    - `gasCost` (BigInt): The estimated gas cost.
+    - `storageCost` (BigInt): The estimated storage cost.
+
+**Example**
+```javascript
+const cost = await ethStorage.estimateCost("dataKey", Buffer.from("some data"));
+console.log(`Gas Cost: ${cost.gasCost}, Storage Cost: ${cost.storageCost}`);
+```
+
+
+
+#### write
+
+**Description**: Asynchronously writes data to the EthStorage network.
+
+**Parameters**
+- `key` (string): The key for the data to be written.
+- `data` (Buffer): The data to be written, its size cannot exceed the maximum value of the content that can be transferred by a blob.
+
+**Returns**
+- `status` (Promise<boolean>): A Promise that resolves to the execution result. `true|false`
+
+**Example**
+```javascript
+const status = await ethStorage.write("dataKey", Buffer.from("some data"));
+```
+
+
+
+#### read
+
+**Description**: Read data asynchronously from the EthStorage network through key.
+
+**Parameters**
+- `key` (string): The key for the data to be read.
+
+**Returns**
+- `data` (Promise<Buffer>): A Promise that resolves to the content.
+
+**Example**
+```javascript
+const data = await ethStorage.read("example.txt");
+```
+
+
+#### putBlobs
+**Description**: Batch upload blob data to the EthStorage network.
+
+**Parameters**
+- `number` (number): Number of blobs.
+- `data` (Buffer): Blob content to be written.
+
+**Returns**
+- `status` (Promise<boolean>): A Promise that resolves to the execution result. `true|false`
+
+**Example**
+```javascript
+const blobData = Buffer.from("some data");
+const status = await ethStorage.putBlobs(number, blobData);
```

Comment:
Is it possible for the SDK to calculate the number for the user based on the blob data?

Reply:
It's more difficult. Why want to do this?

Reply:
Maybe it is also difficult for the user?

Reply:
The SDK only provides support for the contract function `function putBlobs(uint256 num) public payable`.  
The function to calculate `num` should not be placed in the SDK, but where the SDK is called.